stl/inc/algorithm

```diff
@@ -208,18 +208,70 @@ namespace ranges {
 
     // FUNCTION TEMPLATE _Rewrap_subrange
     template <class _Result, class _Wrapped, class _Unwrapped>
-    _NODISCARD constexpr _Result _Rewrap_subrange(
-        [[maybe_unused]] _Wrapped& _First, [[maybe_unused]] subrange<_Unwrapped>&& _UResult) {
-        // conditionally computes a wrapped subrange from a wrapped iterator and unwrapped subrange
-        _STL_INTERNAL_STATIC_ASSERT(is_same_v<_Unwrapped_t<_Wrapped>, _Unwrapped>);
+    _NODISCARD constexpr _Result _Rewrap_subrange(_Wrapped& _Val, subrange<_Unwrapped>&& _UResult) {
+        // conditionally computes a wrapped subrange from a wrapped iterator or range and unwrapped subrange
         if constexpr (is_same_v<_Result, dangling>) {
             return dangling{};
+        } else if constexpr (is_same_v<subrange<_Unwrapped>, _Result>) {
+            return _STD move(_UResult);
+        } else if constexpr (range<_Wrapped>) {
+            _STL_INTERNAL_STATIC_ASSERT(forward_range<_Wrapped>);
+            _STL_INTERNAL_STATIC_ASSERT(is_same_v<_Unwrapped, _Unwrapped_t<iterator_t<_Wrapped>>>);
+            _STL_INTERNAL_STATIC_ASSERT(is_same_v<_Result, subrange<iterator_t<_Wrapped>>>);
+
+            auto _First = _RANGES begin(_Val);
+            auto _Last  = _First;
+            _First._Seek_to(_UResult.begin());
+            _Last._Seek_to(_UResult.end());
+            return _Result{_STD move(_First), _STD move(_Last)};
         } else {
+            _STL_INTERNAL_STATIC_ASSERT(is_same_v<_Unwrapped, _Unwrapped_t<_Wrapped>>);
             _STL_INTERNAL_STATIC_ASSERT(is_same_v<_Result, subrange<_Wrapped>>);
-            auto _Last = _First;
-            _Seek_wrapped(_First, _UResult.begin());
-            _Seek_wrapped(_Last, _UResult.end());
-            return _Result{_STD move(_First), _STD move(_Last)};
+
+            auto _Last = _Val;
+            _Val._Seek_to(_UResult.begin());

```

Comment:
I observe that sometimes we modify `_Val` and sometimes we don't. Does this pose any potential for misuse? (Ok if the answer is "nope" or "very low risk".)

Reply:
This poses very low risk. `_Rewrap_subrange` transforms a subrange of unwrapped iterators - returned from an unchecked helper - into a subrange of wrapped iterators to return to a user. It should therefore always be the last thing an algorithm does, and indeed the existing uses are all in `return` statements. `_Val` is typically move-unwrapped-from on entry to this function and either in the same state or moved-from on return, so the caller doesn't want to use `_Val` after the call in any case.